browser/profiles/brave_renderer_updater.cc

```diff
@@ -47,6 +55,10 @@ BraveRendererUpdater::BraveRendererUpdater(Profile* profile)
 
 BraveRendererUpdater::~BraveRendererUpdater() = default;
 
+void BraveRendererUpdater::UpdateAllRenderersForTesting() {
+  UpdateAllRenderers();
+}
```

Comment:
why do we need this? `BraveRendererUpdater` is spawned per renderer and get updated when constructed.
We just need to make sure pref is set before we navigate or reload.
See `SolanaProviderRendererTest.ExtensionOverwrite`

Reply:
There may already be a precreated renderer when tests starts, so we need to update renderer state manually

Reply:
no need to do that, for testing this, you just need to reload page.

Reply:
and that is exactly what end users would do after changing default wallet setting